### DISCUSSION TITLE
Create DNG from JPGs / TIFFs

### DISCUSSION THREAD

**Discussion by gwegner** [dnglab/dnglab#294] [Ideas]
Adobe Lightroom is currently AFAIK the only software that allows to convert (wrap) JPG and TIFF files into DNG.
Unfortunately they never build that feature into their DNG Converter so that it can't be used in a batch process.
https://community.adobe.com/t5/camera-raw-ideas/p-dng-converter-enable-jpg-conversion/idc-p/13477145/page/6

Would you be open to implement this in DNG Lab? That would be a massive feature!

Of course I'm well aware of the difference between Rawfiles and "Linear" files like JPG/TIFF. Also I'm aware of the fact that wrapping JPG/TIFF into DNG wouldn't create a Raw file.
But: having a Dng would allow us to use the Camera Raw engine to develop previews from the DNG + XMP file which is not possible on JPG's / TIFFs.

Thanks for considering this, of course, I'm open discuss this further!

**Comment by cytrinox** [comment]
Please raise an issue and attach some DNGs, converted from:
 * sRGB JPG source image
 * ProFotoRGB JPG source image
 * Grayscale TIFF 16 bit source image

Basically, converting TIFF to DNG is easy to implement as it's already done because most raw formats are TIFF files internally. There is some metadata checking to choose the correct decoder, but if it is regular TIFF file, the generic tiff reader can be used (that's the fallback not yet implemented).
JPG-to-DNG feels just wrong, as choosing jpg as an intermediate format just to produce a lossless DNG file in the next step is no good workflow. Do you have any good examples for workflows where lossy compressed jpg is a good intermediate step?

The most important question is: are there any restrictions in color spaces that can be embedded in DNG? Usually, the raw pixels are in camera color space and transformed to XYZ space by embedded color matrix. While it's technically possible to put whatever pixel data you have into DNG, I belive there are some situations where math just go wrong. Can such DNGs - derived from sRGB color data - processed in other DNG-capable raw editors while still using correct color science? Just because Adobe has implemented it doesn't mean it's correct in therms of the DNG specification :smile:

**Comment by gwegner** [comment]
Thanks for coming back to me and sorry for the delay in answer.

Let me explain the reason for that DNG from JPGs/TIFFs.

I'm the developer of a software that calculates image edits on basis of XMP files. Those XMP files need to be applied to the Source files in order to get the developed image.
Unfortunately Adobe doesn't provide a real batch process for this, that's why I use the Adobe DNG Converter to create a DNG from the Image File (Raw and DNG works here as input) - this makes the DNG Converter also embed a preview with the developed image. It's all about this preview that I'm using then for visualization of the edits.

Unfortunately, while you can have Lightroom develop JPG/TIFF+XMP in order to get the image developed with the edits in the XMP, the DNG Converter won't alllow JPG/TIFF+XMP to be converted, therefore currently there is no batch option to get a preview for JPGs/TIFFs.

The work around currently is, to bring the JPG sequence into Lighroom and use Lightroom's internal function create a DNG from the JPG/TIFF. Then those DNGs can again be fed into the DNG Converter to get the developed previews.

That process is quite sophisticated, but it works very well since years. The only missing puzzle piece is the support for JPG/TIFF directy (without the manual deroute through Lightroom). 

You see, it doesn't matter if those DNGs are larger than the JPG while having the same quality. They are only intermediaries to be fed into the Adobe DNG Converter in order to produce the preview.

That's why I was so excited to find your solution and though I'd reach out and ask if you would be willing to add that JPG/TIFF support. This would be such a game changer for me! Thanks again for considering.

Here are some sample files. If you need anything else, please let me know!
https://www.dropbox.com/s/ult1gd11jxtcq6h/DNG_from_JPG_TIFF.zip?dl=1

**Comment by cytrinox** [comment]
This is implemented in dnglab 0.6 -> close.
